pdh_windows.go

```diff
@@ -0,0 +1,120 @@
+package sigar
+
+/* These are convenience methods for extracting metrics from the Windows
+   Performance Data Helper (PDH) library */
+
+import (
+	"fmt"
+	"github.com/scalingdata/win"
+	"unsafe"
+)
+
+/* Run the set of provided queries once and report back the raw counter values for each.
+   Queries must have a wildcard, like `\Processor(*)\% idle time`. Returns a map of
+   wildcard values to arrays of results. The indices in the array match up with the
+   index of the query in `queries`. */
+func runRawPdhArrayQueries(queries []string) (map[string][]uint64, error) {
+	var query win.PDH_HQUERY
+	counters := make([]win.PDH_HCOUNTER, len(queries))
+	results := make(map[string][]uint64, len(queries))
+	// Open a new query
+	err := win.PdhOpenQuery(0, 0, &query)
+	if err != 0 {
+		return nil, fmt.Errorf("Error opening PDH query: %X", err)
+	}
+
+	// Add all of the counters. If any fail, give up.
```

Comment:
`PdhAddQuery` is not likely to fail unless the API changes under us and those counters cease to exist (or we run out of memory). I don't think it's worth the added complexity to try and handle partial results. The same goes for when we're fetching the results below - reading raw counters is pretty unlikely to blow up unless something majorly bad is happening.